### PR TITLE
Fix token regression - re-add support for empty tokens

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix token regression - re-add support for empty tokens since this was originally being accessed from `replaceGreetingTokens` which is no longer called by `tokenCompatSubscriber`

Before
----------------------------------------
`{contact.first_name}{ }{contact.last_name}` leaves in `{ }`

After
----------------------------------------
`{contact.first_name}{ }{contact.last_name}`  removes `{ }`

Technical Details
----------------------------------------
I originally thought the behaviour was to remove those tokens when there was a string either side - the tests show that not to be working for `replaceGreetingTokens` - this syncs the behaviour of the 2 rather than trying to improve it.

Comments
----------------------------------------
@totten @colemanw note this test specifically covers the 2 variants of prefix_id - but won't pass without https://github.com/civicrm/civicrm-core/pull/21705 merged

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@totten @colemanw we can merge this as is for parity but I think the before & after are not working as well as I thought - there is still a double space when middlename is empty. I suspect it's clever enough for some cases but not others

File: tests/phpunit/CRM/Utils/TokenConsistencyTest.php [hunk 1]
```diff
@@ -234,6 +234,45 @@ public function getUnadvertisedTokens(): array {
...
+   * 1) as part of the greeting processing
+   * 2) via the token processor.
+   *
+   */
+  public function testOddTokens(): void {
+
+    $variants = [
+      [
+        'string' => '{contact.individual_prefix}{ }{contact.first_name}{ }{contact.middle_name}{ }{contact.last_name}{ }{contact.individual_suffix}',
+        'expected' => 'Mr. Anthony  Anderson II',
```
Comment (totten) [inline tests/phpunit/CRM/Utils/TokenConsistencyTest.php:249]: OK, so your hesitation is that this requires a double-space? Which would be invisible on most HTML outputs - but visible on text outputs?

I think if this produces the same output on 5.42, then I wouldn't worry about it.

**Comment by totten** [comment]
jenkins, test this please

File: tests/phpunit/CRM/Utils/TokenConsistencyTest.php (hunk 1 above)
Comment (eileenmcnaughton) [reply tests/phpunit/CRM/Utils/TokenConsistencyTest.php:249]: thanks @totten

**Pull request merged**